carRental/src/main/java/com/amigoscode/chohort2/carRental/spcefication/Util.java

```diff
@@ -40,9 +40,8 @@ static <T, F> Predicate equal(Root<T> root, CriteriaBuilder cb,
                                   SingularAttribute<? super T, F> attributeName, F input) {
         if (input != null) {
 
-            if (input instanceof Double asNumber) {
-                Expression<Double> trunc = cb.function("trunc", Double.class, root.get(attributeName));
-
+            if (input instanceof Number) {
+                Expression<Number> trunc = cb.function("trunc", Number.class, root.get(attributeName));
                 return cb.equal(trunc, input);
             }
```

Comment:
Just a question: why don't we have to cast ### input here or use pattern matching?

Reply:
no need for casting, I'm not using the input itself in the function  or 'Expression'